xarray/conventions.py

```diff
@@ -165,11 +165,18 @@ def decode_cf_timedelta(num_timedeltas, units):
     """
     num_timedeltas = _asarray_or_scalar(num_timedeltas)
     units = _netcdf_to_numpy_timeunit(units)
+
+    shape = None
+    if isinstance(num_timedeltas, np.ndarray):
```

Comment:
I can change it. The only reason I did this way was to preserve the `_asarray_or_scalar(x)` behavior from above.